pkg/configuration/base/resources/pod.go

```diff
@@ -232,6 +232,11 @@ func NewJenkinsMasterContainer(jenkins *v1alpha2.Jenkins) corev1.Container {
 		envs = append(envs, jenkinsHomeEnvVar)
 	}
 
+	if !IsProxyOverridden(envs) {
+		ProxyEnvVars = dropEmptyProxyEnv(ProxyEnvVars)
+		envs = append(envs, ProxyEnvVars...)
```

Comment:
shouldn't you remove ProxyEnvVars by key instead ?

Reply:
No it is good like this